src/err.rs

```diff
@@ -450,14 +472,25 @@ impl<'a> IntoPy<PyObject> for &'a PyErr {
 
 /// Convert `PyDowncastError` to Python `TypeError`.
 impl std::convert::From<PyDowncastError> for PyErr {
-    fn from(_err: PyDowncastError) -> PyErr {
-        exceptions::TypeError.into()
+    fn from(err: PyDowncastError) -> PyErr {
+        match (err.from, err.to) {
+            (Some(from), Some(to)) => {
+                exceptions::TypeError::py_err(format!("Failed to convert from {} to {}", from, to))
+            }
+            (Some(from), None) => {
+                exceptions::TypeError::py_err(format!("Failed to convert from {}", from))
+            }
+            (None, Some(to)) => {
+                exceptions::TypeError::py_err(format!("Failed to convert to {}", to))
+            }
+            (None, None) => exceptions::TypeError::py_err("Failed to convert"),
+        }
     }
 }
 
 impl<'p> std::fmt::Debug for PyDowncastError {
     fn fmt(&self, f: &mut std::fmt::Formatter) -> Result<(), std::fmt::Error> {
-        f.write_str("PyDowncastError")
+        write!(f, "PyDowncastError from {:?} to {:?}", self.from, self.to)
```

Comment:
This now looks more like a `Display` impl - can you please change this impl to be for `Display` and just add `#[derive(Debug)]` to the `PyDowncastError` struct?